docs: Add tabs to docs where possible #1785

oadeniran opened the pull request:
**The PR fulfills these requirements:** (check all the apply)

- [x] It's submitted to the `main` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `feat: Add a button #xxx`, where "xxx" is the issue number).
- [x] When resolving a specific issue, the PR description includes `Closes #xxx`, where "xxx" is the issue number.
- [ ] If changes were made to `ui` folder, unit tests (`make test`) still pass.
- [ ] New/updated tests are included

My specific changes:
- Tutorial-hello.md renamed to Tutorial-hello.mdx
- edits to both installation.mdx files
- Edited the references to tutorial-hello.md  and redirected to tutorial-hello.mdx (in guide, tutorial-beer, tutorial-counter, tutorial-monitor)
- edited the setup in makefile of tools/showcase from ./venv/bin/python -m playwright install to  ./venv/bin/python -m playwright install-deps
- The other edits and additions were automatically generated

Closes #1785

mturoci commented on website/docs/installation.mdx:
```diff
@@ -16,28 +19,39 @@ pip install h2o-wave
+</TabItem>
 
-To install using Conda:
+<TabItem value="conda" label="Conda">
```
Let's not mix Conda with virtual env section. A better option would be to make a new PyPi/Conda tabs group above. Also do not forget about including the info bar saying wave server is not packaged into Conda  - as conda in virtual env has.

oadeniran replied:
Hi
![Screenshot 2023-07-07 100521](https://github.com/h2oai/wave/assets/60529828/6e41d5c0-132c-435f-9a04-ea224b1f218d)

Does this work for this issue

mturoci replied:
Yes, looking good. One small thing, let's go with `MacOS` instead of `MacOs` everywhere.

oadeniran replied:
Noted

mturoci commented on website/docs/installation.mdx:
```diff
@@ -16,28 +19,39 @@ pip install h2o-wave
-To install in a Conda virtual environment:
+</TabItem>
+
+<TabItem value="condaenv" label="Conda Environment">
```
```suggestion
<TabItem value="condaenv" label="Conda (All OS)">
```

oadeniran replied:
Done

mturoci commented on website/docs/installation.mdx:
```diff
@@ -16,28 +19,39 @@ pip install h2o-wave
 
+<Tabs groupId="operating-systems">
+
+<TabItem value="lin" label="Linux">
```
```suggestion
<TabItem value="lin" label="Linux / MacOS (pip)">
```

mturoci commented on website/docs/installation.mdx:
```diff
@@ -16,28 +19,39 @@ pip install h2o-wave
-To install in a virtual environment (Windows):
+</TabItem>
+
+<TabItem value="win" label="Windows">
```
```suggestion
<TabItem value="win" label="Windows (pip)">
```

mturoci commented on tools/showcase/Makefile:
```diff
@@ -5,7 +5,7 @@ setup:
 	./venv/bin/python -m pip install --upgrade pip
 	./venv/bin/python -m pip install -r requirements.txt
-	./venv/bin/python -m playwright install
+	./venv/bin/python -m playwright install-deps ## was formerly install only 
```
Seems like this is only needed if you didn't have NodeJS 16+ already installed. We have https://github.com/h2oai/wave/blob/main/.nvmrc set properly. Can you try using a regular `./venv/bin/python -m playwright install` with a proper NodeJS version and see if it works for you?

```suggestion
	./venv/bin/python -m playwright install-deps
```

oadeniran replied:
Oh This
will rerun this in a bit and get back to you

oadeniran replied:
![Screenshot 2023-07-07 093013](https://github.com/h2oai/wave/assets/60529828/d5a8cce5-c21a-4677-aa40-2a9493ad0291)

I used the nvrms as instructed and changed my version to 16 but still the same behaviour

mturoci replied:
What does "same behavior" mean? From the screenshot above it seems like you already have the browsers installed.

Can you try using the old Makefile (with `playwright install`)

```
cd tools/showcase
make clean
make setup
make generate
```
and see if it runs properly with Node 16 set?

oadeniran replied:
Same behaviour as in the first line does nothing but the second line initiates the process of installing
I will run the commands you just specified though

mturoci replied:
> Same behaviour as in the first line does nothing

That's fine. It should install the necessary browsers. If you already have them installed, it should do nothing. The important part is if you are able to run `make generate` properly.

oadeniran replied:
So do I revert my change to the makefile then

mturoci replied:
If the steps I outlined above work for you properly (images for docs are generated properly), then yes, revert your changes and let's go with the original Makefile.

oadeniran replied:
Done and committed already

oadeniran replied:
Alright, cleaning and running now

oadeniran replied:
Confirmed 
Using 16+ fixed it 
so good to go
my last commit reverted the makefile

mturoci reviewed (changes requested):
Thanks a lot @oadeniran!

A few minor changes required. I didn't want to duplicate comments for each file so please go through my suggestions and apply them to other files as needed.

When running the Docusaurus docs, I get this warning
```
[WARNING] Docs markdown link couldn't be resolved: (tutorial-hello.md) in "/Users/martinturoci/go/src/github.com/h2oai/telesync/website/docs/scripts.md" for version current
```

oadeniran commented:
Hi, I just Addressed all points you made in the edits and changes commit
I think the reason why the temporary files don't delete is because my "make generate" doesn't run till the end on my wsl

mturoci commented:
> I think the reason why the temporary files don't delete is because my "make generate" doesn't run till the end on my wsl

Why doesn't it run to completion? Btw, the showcase files are still part of this PR, they should not.

> I just Addressed all points you made in the edits and changes commit

There are still comments from me that didn't get answered / resolved.

oadeniran commented:
> > I think the reason why the temporary files don't delete is because my "make generate" doesn't run till the end on my wsl
> 
> Why doesn't it run to completion? Btw, the showcase files are still part of this PR, they should not.
> 
I think it doesn't run till completion due to the inadequacies of my pc. The computing power needed for matplotlib  is a bit high for the project I think and after a while my wsl just gets stuck on generating a file and never gets past that stage
I will remove them from the git history

> > I just Addressed all points you made in the edits and changes commit
> 
> There are still comments from me that didn't get answered / resolved.

oadeniran commented on website/docs/installation.mdx:
```diff
@@ -16,28 +19,39 @@ pip install h2o-wave
-venv\Scripts\activate
+python -m venv venv
+cd venv/Scripts/
+./activate.bat(CMD) or ./Activate.ps1(poweshell)
```
yes
I edited it already
![Screenshot 2023-07-07 101244](https://github.com/h2oai/wave/assets/60529828/5eddb815-35fa-4009-ad35-0cd1e724f007)

Hope it's better

oadeniran commented on py/showcase/showcase1.py:
```diff
@@ -0,0 +1,24 @@
+
```
All deleted

oadeniran commented on website/docs/tutorial-hello.mdx:
```diff
@@ -31,30 +34,68 @@ cd $HOME/wave-apps
 
 ### Install the Wave Python driver
 
+<Tabs groupId="operating-systems">
```
Done

mturoci reviewed (approved):
LGTM. Thanks @oadeniran!

oadeniran commented:
Thanks so much for the approval @mturoci 
I really learned a lot on this
i am thinking of posting this on LinkedIn and I was hoping if its possible for me to tag you on the post to thank you for the help and guidelines provided and also connect with you on linkedin at the same time

mturoci commented:
Sure, np :) Glad you learned something.
